pkg/controller/direct/kms/keyhandle/keyhandle_controller.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/config"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/directbase"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/registry"
+
+	gcp "cloud.google.com/go/kms/apiv1"
+
+	kmspb "cloud.google.com/go/kms/apiv1/kmspb"
+	"google.golang.org/api/option"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ctrlName = "kms-keyhandle-controller"
+)
+
+func init() {
+	registry.RegisterModel(krm.KMSKeyHandleGVK, NewModel)
+}
+
+func NewModel(ctx context.Context, config *config.ControllerConfig) (directbase.Model, error) {
+	return &model{config: *config}, nil
+}
+
+var _ directbase.Model = &model{}
+
+type model struct {
+	config config.ControllerConfig
+}
+
+func (m *model) client(ctx context.Context) (*gcp.AutokeyClient, error) {
+	var opts []option.ClientOption
+	opts, err := m.config.RESTClientOptions()
+	if err != nil {
+		return nil, err
+	}
+	gcpClient, err := gcp.NewAutokeyRESTClient(ctx, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("building KeyHandle client: %w", err)
+	}
+	return gcpClient, err
+}
+
+func (m *model) AdapterForObject(ctx context.Context, reader client.Reader, u *unstructured.Unstructured) (directbase.Adapter, error) {
+	obj := &krm.KMSKeyHandle{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &obj); err != nil {
+		return nil, fmt.Errorf("error converting to %T: %w", obj, err)
+	}
+
+	id, err := krm.NewKMSKeyHandleRef(ctx, reader, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	gcpClient, err := m.client(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &Adapter{
+		id:        id,
+		gcpClient: gcpClient,
+		desired:   obj,
+	}, nil
+}
+
+func (m *model) AdapterForURL(ctx context.Context, url string) (directbase.Adapter, error) {
+	// TODO: Support URLs
+	return nil, nil
+}
+
+type Adapter struct {
+	id        *krm.KMSKeyHandleRef
+	gcpClient *gcp.AutokeyClient
+	desired   *krm.KMSKeyHandle
+	actual    *kmspb.KeyHandle
+}
+
+var _ directbase.Adapter = &Adapter{}
+
+func (a *Adapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx).WithName(ctrlName)
+	log.V(2).Info("getting KeyHandle", "name", a.id.External)
+
+	req := &kmspb.GetKeyHandleRequest{Name: a.id.External}
+	keyhandlepb, err := a.gcpClient.GetKeyHandle(ctx, req)
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting KeyHandle %q: %w", a.id.External, err)
+	}
+
+	a.actual = keyhandlepb
+	return true, nil
+}
+
+func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx).WithName(ctrlName)
+	log.V(2).Info("creating KeyHandle", "name", a.id.External)
```

Comment:
There is no External because the ID is assigned  by the service, you can follow this [example](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/controller/direct/bigqueryconnection/connection_controller.go)